cli/pcluster/utils.py

```diff
@@ -329,6 +329,87 @@ def get_supported_compute_instance_types(scheduler):
     return instances
 
 
+def get_supported_az_for_one_instance_type(instance_type):
+    """
+    Return a tuple of availability zones that have the instance_types.
+
+    This function build above get_supported_az_for_multi_instance_types,
+    but simplify the input to 1 instance type and result to a list
+
+    :param instance_type: the instance type for which the supporting AZs.
+    :return: a tuple of the supporting AZs
+    """
+    return get_supported_az_for_multi_instance_types([instance_type])[instance_type]
+
+
+def get_supported_az_for_multi_instance_types(instance_types):
+    """
+    Return a dict of instance types to list of availability zones that have each of the instance_types.
+
+    :param instance_types: the list of instance types for which the supporting AZs.
+    :return: a dicts. keys are strings of instance type, values are the tuples of the supporting AZs
+
+    Example:
+    If instance_types is:
+    ["t2.micro", "t2.large"]
+    Result can be:
+    {
+        "t2.micro": (us-east-1a, us-east-1b),
+        "t2.large": (us-east-1a, us-east-1b)
+    }
+    """
+    # Most of the fancy code in the func is to try using only one API call for all instance types
+    if not hasattr(get_supported_az_for_multi_instance_types, "cache"):
+        get_supported_az_for_multi_instance_types.cache = {}
+    cache = get_supported_az_for_multi_instance_types.cache
+    missing_instance_types = []
+    result = {}
+    for instance_type in instance_types:
+        if instance_type in cache:
+            result[instance_type] = cache[instance_type]
+        else:
+            missing_instance_types.append(instance_type)
+    if missing_instance_types:
+        ec2_client = boto3.client("ec2")
+        paginator = ec2_client.get_paginator("describe_instance_type_offerings")
+        page_iterator = paginator.paginate(
+            LocationType="availability-zone", Filters=[{"Name": "instance-type", "Values": missing_instance_types}]
+        )
+        offerings = []
+        for page in page_iterator:
+            offerings.extend(page["InstanceTypeOfferings"])
+        for instance_type in missing_instance_types:
+            cache[instance_type] = tuple(
```

Comment:
Minor: Let's try to use a list here instead of tuple?

Reply:
Why list over tuple? I thought tuple is immutable, so better suited for the code here.